### PR TITLE
Improve root cert rotation

### PR DESCRIPTION
**Pull request by JimmyCYJ**
- Remove ```readSigningCertOnly``` from rotator config, and keep rotator behaves the same in read-only Citadel and self-signed Citadel.
- Add a rollback step when root cert rotation fails, to keep the root cert consistently in ```istio-ca-secret```, in memory keycertbundle, and configmap ```istio-security```.
- Fix the update of KCBSyncTime in workload secret controller
- Simplifies the retry logic in casecret.go and configmap.go

#17059

### CODE REVIEW

File: security/pkg/k8s/configmap/configmap.go [hunk 1]
```diff
@@ -84,25 +83,18 @@ func (c *Controller) InsertCATLSRootCert(value string) error {
...
-			if err = c.InsertCATLSRootCert(value); err == nil {
-				break
-			}
-		case <-ctx.Done():
-			configMapLog.Error("Failed to update root cert in config map until timeout.")
-			ticker.Stop()
+	start := time.Now()
+	for {
+		err := c.InsertCATLSRootCert(value)
+		if err == nil {
```
Comment (incfly) [inline security/pkg/k8s/configmap/configmap.go]: log out err otherwise?

Comment (incfly) [reply security/pkg/k8s/configmap/configmap.go]: failed attempts are not loged now.

Comment (JimmyCYJ) [reply security/pkg/k8s/configmap/configmap.go]: Good point. Log is added now.

**Pull request merged**